src/pybind/mgr/dashboard/module.py

```diff
@@ -409,8 +413,8 @@ def handle_command(self, inbuf, cmd):
 
     def create_self_signed_cert(self):
         cert, pkey = create_self_signed_cert('IT', 'ceph-dashboard')
-        self.set_store('crt', cert)
-        self.set_store('key', pkey)
+        self.set_ssl_certificate(inbuf=cert)
+        self.set_ssl_certificate_key(inbuf=pkey)
```

Comment:
What about removing this command from the legacy `handle_command` switch-case and putting it here? And I think we should also check if the previous sub-command failed.

```py
@CLIWriteCommand("dashboard create-self-signed-cert")
def create_self_signed_cert(self):
  cert, pkey = create_self_signed_cert('IT', 'ceph-dashboard')
  r = HandleCommandResult(*self.set_ssl_item('self-signed certificate', 'crt', mgr_id=None))
  if r.retval != 0:
    return r

  r = HandleCommandResult(*self.set_ssl_item('self-signed certificate key', 'key', mgr_id=None))
  if r.retval != 0:
    return r
  return 0, Self-signed certificate created', ''  
```